### PR TITLE
auth-backend: use the catalog client

### PR DESCRIPTION
**Pull request by freben**
Starting to remove all the raw fetches to the catalog

### CODE REVIEW

File: plugins/auth-backend/src/lib/catalog/CatalogIdentityClient.ts
Comment (Rugvip) [inline plugins/auth-backend/src/lib/catalog/CatalogIdentityClient.ts:58]: This shouldn't be needed should it? `Entity` should be a subset of `UserEntity`

**Review by Rugvip — APPROVED** [review]
:shipit:, but a nit

File: plugins/auth-backend/src/lib/catalog/CatalogIdentityClient.ts
Comment (freben) [reply plugins/auth-backend/src/lib/catalog/CatalogIdentityClient.ts]: correct, fix'd

won't merge until the other PR that this is on top of is merged tho

**Review by hooloovooo — APPROVED** [review]

**Pull request merged**
